pallets/parachain-staking/src/rewards/transfer_from_rewards_account.rs

```diff
@@ -0,0 +1,212 @@
+			Error::<Runtime>::DeadAccount
+		);
+
+		// Directly handle the result of the transfer, making use of match
+		// for clearer error handling.
+		match Runtime::Currency::transfer(
+			&rewards_account,
+			&delegator_id,
+			amount,
+			ExistenceRequirement::KeepAlive,
+		) {
+			Ok(_) => Ok(amount),
+			Err(DispatchError::Arithmetic(ArithmeticError::Underflow)) => Ok(Zero::zero()),
+			Err(e) => Err(e),
+		}
+	}
+}
+
+impl<T: Config> Pallet<T> {
+	pub fn send_collator_reward(
+		_round_idx: RoundIndex,
+		collator_id: T::AccountId,
+		amt: BalanceOf<T>,
+	) -> Weight {
+		// Check if the collator's account exists; return early if not.
+		if !frame_system::Account::<T>::contains_key(&collator_id) {
+			return Weight::zero(); // TODO
+		}
+
+		// Attempt to get the RewardsAccount and return early if not set.
+		let rewards_account = match RewardsAccount::<T>::get() {
+			Some(account) => account,
+			None => {
+				return Weight::zero(); // TODO Adjust with the actual weight for a missing rewards account.
+			},
+		};
+
+		// Proceed with the transfer and handle the result.
+		let transfer_result = T::Currency::transfer(
+			&rewards_account,
+			&collator_id,
+			amt,
+			ExistenceRequirement::KeepAlive,
+		);
+
+		match transfer_result {
+			Ok(_) => {
+				Self::deposit_event(Event::Rewarded { account: collator_id, rewards: amt });
+			},
+			Err(e) =>
+				log::error!("💥 Failed to send reward to collator: {:?}, amount: {:?}", e, amt),
+		}
+
+		Weight::zero() // TODO
+	}
+}
+
+#[cfg(test)]
+mod tests {
+	use super::*;
+	use crate::mock::*;
+	use frame_support::{assert_err, assert_ok};
+
+	#[test]
+	fn test_payout_collator_without_rewards_account() {
+		ExtBuilder::default().build().execute_with(|| {
+			let collator = 10;
+			let amount = 8;
+			let round_index = 0;
+
+			let _ = pallet_balances::Pallet::<Test>::deposit_creating(&collator, 1);
+
+			<TransferFromRewardsAccount as PayoutReward<Test>>::payout_collator_rewards(
+				round_index,
+				collator,
+				amount,
+			);
+
+			assert_eq!(pallet_balances::Pallet::<Test>::free_balance(&collator), 1);
+
+			assert_no_events!();
+		});
+	}
+
+	#[test]
+	fn test_payout_collator_with_not_enough_funds_in_rewards_account() {
+		ExtBuilder::default().with_rewards_account(999, 7).build().execute_with(|| {
+			let collator = 10;
+			let amount = 8;
+			let round_index = 0;
+
+			let _ = pallet_balances::Pallet::<Test>::deposit_creating(&collator, 1);
+
+			<TransferFromRewardsAccount as PayoutReward<Test>>::payout_collator_rewards(
+				round_index,
+				collator,
+				amount,
+			);
+
+			assert_eq!(pallet_balances::Pallet::<Test>::free_balance(&collator), 1);
+
+			assert_no_events!();
+		});
+	}
+
+	#[test]
+	fn test_payout_with_no_rewards_account_should_do_nothing() {
+		ExtBuilder::default().build().execute_with(|| {
+			let delegator = 0;
+			let amount = 100;
+
+			let _ = pallet_balances::Pallet::<Test>::deposit_creating(&delegator, 1);
+
+			assert_ok!(
```

Comment:
probably better to use `assert_noop` here

Reply:
`asset_noop` is used in error cases

Reply:
ah yeah, then `assert_storage_noop`